### PR TITLE
(DOCSP-27496): C++: Add Map docs, clean up embedded object docs

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

The work for embedded object cleanup was minimal, so I combined it with adding docs for Map - this work is mainly about the Map docs.

- https://jira.mongodb.org/browse/DOCSP-27496
- https://jira.mongodb.org/browse/DOCSP-27947

### Staged Changes

- [Supported Data Types](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27947/sdk/cpp/model-data/supported-types/#map-dictionary): New section for Map, new supported property declaration example of map
- [Create](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27947/sdk/cpp/crud/create/#create-an-object-with-a-map-property): New section for creating an object with a map property
- [Read](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27947/sdk/cpp/crud/read/#read-a-map-property): New section for reading a map property
- [Update](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27947/sdk/cpp/crud/update/#overwrite-an-embedded-object-property): New sections for updating a map property, overwriting an embedded object property
- [Delete](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-27947/sdk/cpp/crud/delete/#delete-map-keys-values): New section for deleting a map key

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/cpp/define-object-model.cpp [hunk 1]
```diff
@@ -51,6 +53,25 @@ struct Employee : realm::object<Employee> {
 };
 // :snippet-end:
 
+// :snippet-start: model-with-map-property
+struct Map_Employee : realm::object<Map_Employee> {
+    enum class WorkLocation {
+        home, office
```
Comment (cbush) [inline examples/cpp/define-object-model.cpp:59]: Conventionally, these would be ALL_CAPS

File: examples/cpp/define-object-model.cpp [hunk 2]
```diff
@@ -373,11 +296,180 @@ TEST_CASE("update an embedded object", "[model][update]") {
...
+}
+
+TEST_CASE("test enum map object", "[model][write]") {
+    // :snippet-start: create-map-object
+    auto realm = realm::open<Map_Employee>();
+
+    auto employee = Map_Employee {
+        ._id = 8675309,
+        .firstName = "Tommy",
+        .lastName = "Tutone"
```
Comment (cbush) [inline examples/cpp/define-object-model.cpp:355]: Haha

I really did know a Jenny with that number (867 was a common prefix in Ottawa)

File: examples/cpp/define-object-model.cpp [hunk 3]
```diff
@@ -373,11 +296,180 @@ TEST_CASE("update an embedded object", "[model][update]") {
...
+        auto dogsNamedMaui = dogs.where([](auto &dog) {
+            return dog.name == "Maui";
+        });
+        REQUIRE(dogsNamedMaui.size() >= 1);
+        auto maui = dogsNamedMaui[0];
+        for (auto [k, v] : maui.favoriteParkByCity) {
+            if (k == "Boston") CHECK(v == "Fort Point");
+            else if (k == "New York") CHECK(v == "Central Park");
+        }
+        auto favoriteBostonPark = maui.favoriteParkByCity["Boston"];
```
Comment (cbush) [inline examples/cpp/define-object-model.cpp:457]: Just one thing to note here is that `operator[]` might create an entry -- usually you would use `find()` for read only access

**Review by cbush — APPROVED** [review]

File: examples/cpp/define-object-model.cpp (hunk 3 above)
Comment (dacharyc) [reply examples/cpp/define-object-model.cpp:457]: Ah! Ok, updating this example. But I'm not actually _using_ this example in the docs - just wanted something simpler (string/string) to validate my understanding. Do you think that's worth noting in the docs, though, or is this something that C++ devs should already know?

Comment (cbush) [reply examples/cpp/define-object-model.cpp:457]: That's something C++ devs would already know.

**Pull request merged**